### ISSUE TITLE
Writing new files is subject to race conditions

### ISSUE THREAD

**Issue by atamon** [Reposoft/repos-upload#3]
If two or more files are to be created at the same time within the same non-existing folder, there's a race condition between checking whether the folder exists and then creating it. 

We should implement an end-to-end test using docker(-compose) for this.

**Comment by solsson** [comment]
> We should implement an end-to-end test using docker(-compose) for this.

Good idea. Ideas for a docker-compose.yml can probably be found in https://github.com/Reposoft/docker-repos-indexing and https://github.com/Reposoft/docker-svn/tree/master/rweb. There's still no official image for https://github.com/Reposoft/rweb.

**Comment by solsson** [comment]
We might want to re-frame this issue: writeFile isn't robust to transient backend failures. Let's discuss something like https://github.com/segmentio/superagent-retry#retrying-cases.
